File: src/main/java/com/freya02/bot/docs/CachedField.java
package com.freya02.bot.docs;

import com.freya02.docs.SeeAlso;
import net.dv8tion.jda.api.entities.MessageEmbed;

import java.util.List;

public final class CachedField {
	private final MessageEmbed fieldEmbed;
	private final List<SeeAlso.SeeAlsoReference> seeAlsoReferences;

	public CachedField(MessageEmbed fieldEmbed,
	                   List<SeeAlso.SeeAlsoReference> seeAlsoReferences) {
		this.fieldEmbed = fieldEmbed;
		this.seeAlsoReferences = seeAlsoReferences;
	}

	public MessageEmbed getFieldEmbed() {
		return fieldEmbed;
	}

	public List<SeeAlso.SeeAlsoReference> getSeeAlsoReferences() {
		return seeAlsoReferences;
	}
}

File: doxxy-base/src/main/kotlin/com/freya02/bot/docs/cached/CachedField.kt
package com.freya02.bot.docs.cached

import com.freya02.docs.DocSourceType
import com.freya02.docs.data.SeeAlso.SeeAlsoReference
import net.dv8tion.jda.api.entities.MessageEmbed

class CachedField(
    override val source: DocSourceType,
    override val embed: MessageEmbed,
    override val seeAlsoReferences: List<SeeAlsoReference>,
    override val javadocLink: String?,
    override val sourceLink: String?
) : CachedDoc